source/common/http/conn_manager_utility.cc

```diff
@@ -77,13 +77,27 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
     }
     request_headers.insertForwardedProto().value().setReference(
         connection.ssl() ? Headers::get().SchemeValues.Https : Headers::get().SchemeValues.Http);
+
+    // Since useRemoteAddress() is true, skip adding x-forwarded-port when the x-forwarded-port
+    // is already set.
+    if (config.appendXForwardedPort() && !request_headers.ForwardedPort()) {
+      // Otherwise, set x-forwarded-for with remote address connection port, i.e. the port of the
+      // proxy in front of this Envoy that is used by the client to connect to.
+      request_headers.insertForwardedPort().value(connection.remoteAddress()->ip()->port());
+    }
   } else {
     // If we are not using remote address, attempt to pull a valid IPv4 or IPv6 address out of XFF.
     // If we find one, it will be used as the downstream address for logging. It may or may not be
     // used for determining internal/external status (see below).
     auto ret = Utility::getLastAddressFromXFF(request_headers, xff_num_trusted_hops);
     final_remote_address = ret.address_;
     single_xff_address = ret.single_address_;
+
+    if (config.appendXForwardedPort()) {
```

Comment:
In this case I think we don't do anything? Or possibly append only if it doesn't exist?

In general I think we should update the docs for x-forwarded-for to also talk about XFP (or at least link to it) since this is complicated and is the same general topic? Also can we make the XFP docs more explicit like the XFF docs to talk about how remote address interacts?